plugins/fileservices/fileservices.cpp

```diff
@@ -2644,4 +2648,33 @@ FILESERVICES_API char * FILESERVICES_CALL fsGetEspURL(const char *username, cons
     return strdup("");
 }
 
+FILESERVICES_API char * FILESERVICES_CALL fsGetDefaultDropZone(const char *username, const char *userPW)
+{
+    Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory(true);
+    Owned<IConstEnvironment> constEnv = envFactory-> openEnvironment();
+    Owned<IConstDropZoneInfoIterator> dropZoneIt = constEnv->getDropZoneIterator();
+    SCMStringBuffer dropZoneDir;
+    if (dropZoneIt->first())
+        dropZoneIt->query().getDirectory(dropZoneDir);
+
+    return strdup(dropZoneDir.str());
+
+}
+
+FILESERVICES_API void FILESERVICES_CALL fsGetDropZones(ICodeContext *ctx, size32_t & __lenResult, void * & __result, const char *username, const char *userPW)
+{
+    MemoryBuffer mb;
+    Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory(true);
+    Owned<IConstEnvironment> constEnv = envFactory-> openEnvironment();
+    Owned<IConstDropZoneInfoIterator> dropZoneIt = constEnv->getDropZoneIterator();
+    ForEach(*dropZoneIt)
+    {
+        SCMStringBuffer dropZoneDir;
+        dropZoneIt->query().getDirectory(dropZoneDir);
+        size32_t sz = dropZoneDir.length();
+        mb.append(sz).append(sz,dropZoneDir.str());
```

Comment:
isn't this concatenating them altogether without separators, such that the caller wouldn't be able to tell where new one begins?

Reply:
it may be better to return as a dataset, 1 row per dropzone

Reply:
There is a "separator", because the mb content is : <len1><record1><len2><record2>...
@ghalliday Is this the right format a dataset returned by a File Service function?

Reply:
Sorry I misread, it is returning a dataset(FsDropZoneRecord).
Does it output a sensible result when there are multiple DropZones defined?

I think format is ok, but will let @ghalliday confirm.

Reply:
The code and format both look sensible.